app/controllers/public/sessions.js

```diff
@@ -17,24 +17,22 @@ export default class SessionsController extends Controller {
   @computed('model.session.@each', 'timezone')
   get groupByDateSessions() {
     let sessions;
-
     if (this.sort === 'title') {
       sessions = groupBy(this.model.session.toArray(), '');
     } else {
       sessions = groupBy(this.model.session.toArray(), s => moment.tz(s.startsAt, this.timezone).format('dddd, Do MMMM'));
     }
-
     return Object.entries(sessions).map(([date, sessions]) => ({ date: date === 'undefined' ? null : date, sessions }));
   }
 
-  @computed('model.event.startsAt', 'model.event.endsAt', 'timezone')
+  @computed('model.dates', 'timezone')
   get allDates() {
-    const arr = [];
-    const difference = (this.model.event.endsAt).diff(this.model.event.startsAt, 'days');
-    for (let i = 0; i <= Math.abs(difference); i++) {
-      arr.push(moment.tz(this.model.event.startsAt, this.timezone).add(i, 'days').toISOString());
-    }
-    return arr;
+
+    const uniqueDates = new Set();
+
+    this.model.dates.toArray().map(el => moment(moment.tz(el.startsAt, this.timezone)).format('YYYY-MM-DD')).sort().map(el => uniqueDates.add(el));
```

Comment:
```suggestion
    const uniqueDates = new Set(this.model.dates.toArray().map(date => moment.tz(date.startsAt, this.timezone)).sort().map(date => moment().format('YYYY-MM-DD')));
```

How hard was this?

I'm reviewing it for 2 days. More than 26 commits later, it is still not at desired stage

Reply:
sir , i literally thought , it would be not readable , that's why i wrote the code in full length from the very beginning.

Reply:
Sir you wrote the code full code, i don't want this pr to be fall in my contributions, thank you , i will find another issue to solve completely.